alvr/common/src/dashboard/mod.rs

```diff
@@ -0,0 +1,112 @@
+mod basic_components;
+mod components;
+
+use basic_components::{modal, ModalResponse};
+use components::{connections_tab, statistics_tab};
+use egui::{Align, CentralPanel, ComboBox, CtxRef, Layout, SidePanel};
+
+#[derive(Clone, Copy, PartialEq, Debug)]
+enum Tab {
+    Connections,
+    Statistics,
+    Settings,
+    Installation,
+    Logs,
+    About,
+}
+
+impl Default for Tab {
+    fn default() -> Self {
+        Tab::Connections
+    }
+}
+
+#[derive(Default)]
+struct LanguageModalState {
+    visible: bool,
+    selection: String,
+}
+
+#[derive(Default)]
+pub struct Dashboard {
+    tab: Tab,
+    language_modal_state: LanguageModalState,
+    language: String,
+}
+
+impl Dashboard {
+    pub fn draw(&mut self, ctx: &CtxRef) {
+        SidePanel::left("side_panel").show(ctx, |ui| {
+            ui.heading("ALVR");
+            egui::warn_if_debug_build(ui);
+
+            for tab in [
+                Tab::Connections,
+                Tab::Statistics,
+                Tab::Settings,
+                Tab::Installation,
+                Tab::Logs,
+                Tab::About,
+            ] {
+                ui.selectable_value(&mut self.tab, tab, format!("{:?}", tab));
+            }
+
+            // ui.with_layout(Layout::bottom_up(Align::LEFT), |ui| {
+            //     if ui.selectable_label(false, "Language").clicked() {
+            //         self.language_modal_state = LanguageModalState {
+            //             visible: true,
+            //             selection: self.language.clone(),
+            //         };
+            //     }
+            // });
+
+            if self.language_modal_state.visible {
+                language_modal(ctx, &mut self.language_modal_state, &mut self.language);
+            }
+        });
+
+        CentralPanel::default().show(ctx, |ui| {
+            ui.heading(format!("{:?}", self.tab));
+            match self.tab {
+                Tab::Connections => connections_tab(ui),
+                Tab::Statistics => statistics_tab(ui),
+                _ => (),
+            }
+        });
+    }
+}
+
+fn language_display(code: &str) -> &str {
+    match code {
+        "en" => "English",
+        "it" => "Italiano",
+        _ => "",
+    }
```

Comment:
Use the `isolang` crate or something like it?

https://lib.rs/crates/isolang (GitHub did not want to render markdown consistently here..)

Reply:
This is just some sketched code just for testing, I will replace this with Fluent (like in the new-dashboard branch).